app/models/user_search.rb

```diff
@@ -13,19 +13,19 @@ def initialize(term, opts = {})
     @searching_user = opts[:searching_user]
     @include_staged_users = opts[:include_staged_users] || false
     @limit = opts[:limit] || 20
-    @group = opts[:group]
+    @groups = opts[:groups]
     @guardian = Guardian.new(@searching_user)
-    @guardian.ensure_can_see_group!(@group) if @group
+    @groups.each { |group| @guardian.ensure_can_see_group!(group) } if @groups
```

Comment:
This is going to hit us with multiple queries. Can you see if there is a possibility of introducing a `can_see_groups` method that does this more efficiently seeing that we've already loaded all the groups into memory.